src/pybind/mgr/dashboard/controllers/service.py

```diff
@@ -45,7 +94,46 @@ def get(self, service_name: str) -> List[dict]:
         services = orch.services.get(service_name)
         if not services:
             raise cherrypy.HTTPError(404, 'Service {} not found'.format(service_name))
-        return services[0].to_json()
+
+        service = services[0].to_json()
+        service_type = service.get('service_type', '')
+
+        # Get all certificates at once
+        cert_ls_data = None
+        try:
+            logger.debug('Calling cert_ls for service %s with filter_by="", '
+                         'show_details=True, include_cephadm_signed=True',
+                         service_name)
+            cert_ls_result = orch.cert_store.cert_ls(
+                filter_by='',
+                show_details=True,
+                include_cephadm_signed=True
+            )
+            cert_ls_data = cert_ls_result if cert_ls_result else {}
+            logger.debug('cert_ls for service %s returned %d certificate(s): %s',
+                         service_name,
+                         len(cert_ls_data) if cert_ls_data else 0,
+                         list(cert_ls_data.keys()) if cert_ls_data else [])
+        except RuntimeError as e:
+            logger.warning('Failed to retrieve certificate list for service %s: %s',
+                           service_name, str(e), exc_info=True)
+            cert_ls_data = {}
```

Comment:
At least you can set it to `{}` by default so you don't need to remap from `None` to `{}`